pyEpiabm/pyEpiabm/tests/test_sweep/test_initialise_place_sweep.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from unittest import mock
+
+import pyEpiabm as pe
+from pyEpiabm.property.infection_status import InfectionStatus
+from pyEpiabm.property.place_type import PlaceType
+
+
+class TestInitialisePlaceSweep(unittest.TestCase):
+    """Test the 'UpdatePlaceSweep' class.
+    """
+
+    def setUp(self) -> None:
+        """Initialises a population with one infected person. Sets up a
+        single household containing this person.
+        """
+        self.pop = pe.Population()
+        self.pop.add_cells(1)
+        self.cell = self.pop.cells[0]
+        self.pop.cells[0].add_microcells(1)
+        self.microcell = self.cell.microcells[0]
+        self.pop.cells[0].microcells[0].add_people(1)
+        self.person = self.pop.cells[0].microcells[0].persons[0]
+        self.person.update_status(InfectionStatus.InfectMild)
+        self.microcell.add_place(1, (1, 1), PlaceType.Workplace)
+        self.place = self.cell.places[0]
+        pe.Parameters.instance().time_steps_per_day = 1
+        self.time = 1
+        self.params = pe.Parameters.instance().place_params
+
+    def test_bind(self):
+        """Tests that the update place sweep correctly binds
+        the given population.
+        """
+        test_pop = self.pop
+        test_sweep = pe.sweep.InitialisePlaceSweep()
+        test_sweep.bind_population(test_pop)
+        self.assertEqual(test_sweep._population.cells[0].
+                         places[0].place_type, pe.property.PlaceType.Workplace)
+
+    @mock.patch("pyEpiabm.sweep.InitialisePlaceSweep.create_age_weights")
+    @mock.patch("pyEpiabm.sweep.UpdatePlaceSweep.update_place_group")
+    def test__call__(self, mock_update, mock_weights):
+        """Test whether the update place sweep function takes an
+        initially empty place and correctly adds a person to
+        the place.
+        """
+        test_pop = self.pop
+        place = test_pop.cells[0].places[0]
+        place.place_type = PlaceType.Workplace
+        person = test_pop.cells[0].persons[0]
+        mock_update.return_value = None
+        mock_weights.return_value = [[person], [1]]
+
+        test_sweep = pe.sweep.InitialisePlaceSweep()
+        test_sweep.bind_population(test_pop)
+        test_sweep()
+        mock_update.called_with(place, [person], [1], 14.28, 5927, 0)
```

Comment:
I have never seen this before, are you sure you don't mean `assert_called_with()`?

Reply:
possibly